arcjet-astro/index.ts

```diff
@@ -0,0 +1,482 @@
+import type {
+  BotOptions,
+  EmailOptions,
+  FixedWindowRateLimitOptions,
+  ProtectSignupOptions,
+  SensitiveInfoOptions,
+  ShieldOptions,
+  SlidingWindowRateLimitOptions,
+  TokenBucketRateLimitOptions,
+} from "arcjet";
+import type { AstroIntegration } from "astro";
+import { z, type ZodType, type ZodTypeDef } from "astro/zod";
+import fs from "node:fs/promises";
+
+const resolvedVirtualClientId = "\0ARCJET_VIRTUAL_CLIENT";
+
+const validateMode = z.enum(["LIVE", "DRY_RUN"]);
+const validateProxies = z.array(z.string());
+const validateCharacteristics = z.array(z.string());
+const validateClientOptions = z
```

Comment:
Do you think it makes sense to try and re-use the validators from the `arcjet` package here where possible? I'm just asking because it would be nice to minimize the number of packages that require code changes when doing feature work, currently we don't really need to update the adapters all that much.

An alternative could be to move those adapters in to a separate adapters package so that they don't need to be publicly exported from the main sdk. Or if we called it an sdk utils package we could put that `errorMessage()` function in there too.